Misc ginkgo cleanups

joestringer opened the pull request:
Sprinkling of various ginkgo/ci-related cleanups:
* Fix panic error message to be more descriptive, useful
* Attempt to gather endpoint info for all cilium pods even if one cilium pod cannot be reached
* Consistently use `time` package for time durations throughout `test/`

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/6576)
<!-- Reviewable:end -->

joestringer committed:
helpers: Use proper time types

Several places in the test code are using times as int64 values rather
than using the native golang types for dealing with time durations. Fix
these up and replace all of the existing hard-coded timeouts with
variables which should make them easier to adjust in future.

Signed-off-by: Joe Stringer <joe@cilium.io>

joestringer committed:
helpers: Collect errors when EndpointWaitReady fails

Previously only one cilium instance would be attempted to be contacted,
and if unavailable, an error would be returned. In case the other Cilium
instance is actually available, collect the error from the first
instance then attempt to continue and print the state for the endpoints
in the other instance.

Signed-off-by: Joe Stringer <joe@cilium.io>

joestringer committed:
helpers: Fix function name in panic

This line was copy-pasted from a similar function and the error message
was not updated. Fix it to reflect where the panic occurs.

Signed-off-by: Joe Stringer <joe@cilium.io>

joestringer committed:
helpers: Limit ep gathering to 5m for all cilium pods

Rather than allowing a limit of 5 minutes per cilium pod for gathering
the endpoints list, restrict it to 5 minutes for all cilium pods.
Existing test cases only ever use up to 2 cilium instances so this still
leaves a reasonable amount of time.

This doesn't have much impact with the current implementation which
terminates at the first error while fetching endpoint status from cilium
instances, but an upcoming commit will change this so we change it here.

Signed-off-by: Joe Stringer <joe@cilium.io>

eloycoto reviewed (approved):
awesome as normal :-)

Commit:
examples: remove blank line at end of dns.yaml

This caused the policy validation CI test to fail.

Signed-off by: Ian Vernon <ian@cilium.io>

ianvernon commented on test/k8sT/Tunnels.go:
```diff
@@ -151,14 +151,14 @@ var _ = Describe("K8sTunnelTest", func() {
 
 func isNodeNetworkingWorking(kubectl *helpers.Kubectl, filter string) bool {
-	err := kubectl.WaitforPods(helpers.DefaultNamespace, fmt.Sprintf("-l %s", filter), 3000)
+	err := kubectl.WaitforPods(helpers.DefaultNamespace, fmt.Sprintf("-l %s", filter), helpers.HelperTimeout)
```
I wonder if this was supposed to be 300 before, because it was 3000. That would mean that it would take 50 minutes for node networking to be ready ;)

joestringer replied:
Yeah, I noticed that too. I considered calling it out in a separate commit but it looks like an obvious typo, 50 minutes is longer than the entire typical CI run.
